File: src/main/java/com/skillstorm/spartanwireless/models/Device.java
package com.skillstorm.spartanwireless.models;

import java.util.List;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.OneToMany;
import jakarta.persistence.Table;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@Getter
@Setter
@Builder
@AllArgsConstructor
@NoArgsConstructor
@Entity
@Table(name = "device")
public class Device {

    @Id
    @Column(name = "device_id")
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long deviceId;

    @Column(name = "name")
    private String name;

    @Column(name = "brand")
    private String brand;

    @Column(name = "price")
    private double price;

    @OneToMany(mappedBy = "primaryKey.device")
    private List<PhoneLine> phoneLines;
}

File: front_end/src/app/models/device.ts

export class Device {

    deviceId: number | 0;
    name: string;
    brand: string;
    price: number;

    constructor(deviceId: number,
                name: string,
                brand: string,
                price: number) {

        this.deviceId = deviceId;
        this.name = name;
        this.brand = brand;
        this.price = price;
    }

    setDeviceId(deviceId: number) {
        this.deviceId = deviceId;
    }

}
